Sources/Plasma/CoreLib/HeadSpin.cpp

```diff
@@ -487,165 +487,98 @@ char    *hsWStringToString( const wchar_t *str )
+                versionStrs.push_back("Personal ");
+            else
+                versionStrs.push_back("Professional ");
         }
-        else
+        else if ( version.wProductType == VER_NT_SERVER )
         {
-            HKEY hKey;
-            char szProductType[80];
-            DWORD dwBufLen;
-
-            RegOpenKeyEx( HKEY_LOCAL_MACHINE,
-                "SYSTEM\\CurrentControlSet\\Control\\ProductOptions",
-                0, KEY_QUERY_VALUE, &hKey );
-            RegQueryValueEx( hKey, "ProductType", NULL, NULL,
-                (LPBYTE) szProductType, &dwBufLen);
-            RegCloseKey( hKey );
-            if ( lstrcmpi( "WINNT", szProductType) == 0 )
-                versionStrs.push_back("Professional ");
-            if ( lstrcmpi( "LANMANNT", szProductType) == 0 )
-                versionStrs.push_back("Server ");
-            if ( lstrcmpi( "SERVERNT", szProductType) == 0 )
+            if( version.wSuiteMask & VER_SUITE_DATACENTER )
+                versionStrs.push_back("DataCenter Server ");
+            else if( version.wSuiteMask & VER_SUITE_ENTERPRISE )
                 versionStrs.push_back("Advanced Server ");
+            else
+                versionStrs.push_back("Server ");
         }
 
         // Display version, service pack (if any), and build number.
 
-        if ( osvi.dwMajorVersion <= 4 )
+        if ( version.dwMajorVersion <= 4 )
         {
             versionStrs.push_back(ST::format("version {}.{} {} (Build {})\n",
-                osvi.dwMajorVersion,
-                osvi.dwMinorVersion,
-                osvi.szCSDVersion,
-                osvi.dwBuildNumber & 0xFFFF));
+                version.dwMajorVersion,
+                version.dwMinorVersion,
+                version.szCSDVersion,
+                version.dwBuildNumber & 0xFFFF));
         }
         else
         {
             versionStrs.push_back(ST::format("{} (Build {})\n",
-                osvi.szCSDVersion,
-                osvi.dwBuildNumber & 0xFFFF));
+                version.szCSDVersion,
+                version.dwBuildNumber & 0xFFFF));
         }
         break;
-
-    case VER_PLATFORM_WIN32_WINDOWS:
-
-        if (osvi.dwMajorVersion == 4 && osvi.dwMinorVersion == 0)
-        {
-            versionStrs.push_back("Microsoft Windows 95 ");
-            if ( osvi.szCSDVersion[1] == 'C' || osvi.szCSDVersion[1] == 'B' )
-                versionStrs.push_back("OSR2 ");
-        }
-
-        if (osvi.dwMajorVersion == 4 && osvi.dwMinorVersion == 10)
-        {
-            versionStrs.push_back("Microsoft Windows 98 ");
-            if ( osvi.szCSDVersion[1] == 'A' )
-                versionStrs.push_back("SE ");
-        }
-
-        if (osvi.dwMajorVersion == 4 && osvi.dwMinorVersion == 90)
-        {
-            versionStrs.push_back("Microsoft Windows Me ");
-        }
-        break;
-
-    case VER_PLATFORM_WIN32s:
-
-        versionStrs.push_back("Microsoft Win32s ");
```

Comment:
:trollface:

Reply:
Pretty much.  I don't think we had a pressing need to keep that around.